advanced/wallets/react-wallet-v2/src/pages/session.tsx

```diff
@@ -35,6 +36,20 @@ export default function SessionPage() {
 
   // Get necessary data from session
   const expiryDate = useMemo(() => new Date(session?.expiry! * 1000), [session])
+  const getPendingRequests = useCallback(() => {
+    if (!session) return
+    const allPending = web3wallet.getPendingSessionRequests()
+    const requestsForSession = allPending?.filter(r => r.topic === session.topic)
+    setPendingRequests(requestsForSession)
+  }, [session])
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      getPendingRequests()
+    }, 1000)
+    getPendingRequests()
+    return () => clearInterval(interval)
+  }, [getPendingRequests])
```

Comment:
why don't we have events for this?

Reply:
we do, but this bundles a couple of functionalities in single command (the interval)
- refreshes the list when when new request comes in
- refreshes the list when request is responded to
- refreshes the expiry timer
any other approach would be more lengthy/complex to develop